src/routes/query.py

```
@@ -0,0 +1,23 @@
+from fastapi import APIRouter, Depends
+from synapso_core.query_manager import QueryManager
+
+from src.models import QueryRequest, QueryResponse
+
+router = APIRouter(prefix="/query", tags=["query"])
+
+
+@router.post("/query", status_code=HTTPStatus.OK, response_model=QueryResponse)
+async def query(
+    request: QueryRequest,
+    query_manager: QueryManager = Depends(QueryManager),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)